solutions/usaco-894.mdx

```diff
@@ -146,41 +169,169 @@ void setIO(string name = ""){ // FastIO see General -> Fast Input and Output
 	}
 }
 
-bool visited[MAX_N + 1];
-int ans[MAX_N + 1];
-vector<int> adjList[MAX_N + 1];
-
-void dfs(int self, int parent){
-	int len = adjList[self].size();
-	int g = 1;
-	for (int i : adjList[self]) {
-		if (i != parent) {
-			while (g == ans[self] || g == ans[parent]) 
-				g++;
-			ans[i] = g;
-			dfs(i, self);
-			g++;
-		}
-	}
-}
-
 int main() {
 	setIO("planting");
 
 	int n; 
 	cin >> n;
+	int deg[MAX_N + 1] = {}; // initialize with 0s
 	int u, v;
-	for (int i = 1; i < n; i++){
+	int maxDeg = INT_MIN;  // tracking maxDegree
+	for (int i = 1; i < n; i++) {
 		cin >> u >> v;
-		adjList[u].push_back(v);
-		adjList[v].push_back(u);
+		deg[u]++;
+		deg[v]++;
+		maxDeg = max(maxDeg, max(deg[u], deg[v]));
 	}
-	dfs(1, 1);
-	int maxAns = 0;
-	for (int curAns : ans) {
-		maxAns = max(maxAns, curAns);
+	cout << maxDeg + 1 << "\n";
+}
+```
+
+</CPPSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class GrassPlanting {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("planting");
+		int n = io.nextInt();
+		int[] deg = new int[n+1];
+		for(int i = 0; i < n-1; i++) {
+			int a = io.nextInt(), b = io.nextInt();
+			deg[a]++; deg[b]++;
+		}
+
+		int ret = 0;
+
+		for(int i = 1; i <= n; i++) ret = Math.max(ret, deg[i]);
+		io.println(ret+1);
+		io.close();
+	}
+
+	private static class Kattio extends PrintWriter {
```

Comment:
you can just use `//CodeSnip{Kattio}` i think

Reply:
how does that work

Reply:
just replace the entire class with 
```java
	//CodeSnip{Kattio}
```